opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1346,6 +1347,16 @@ namespace Opm {
 
         // We use the rates from the privious time-step to reduce oscilations
         WellGroupHelpers::updateWellRates(fieldGroup, schedule(), reportStepIdx, previous_well_state_, well_state_);
+
+        // Set ALQ for off-process wells to zero
+        for (const auto& wname : schedule().wellNames(reportStepIdx)) {
+            const bool is_producer = schedule().getWell(wname, reportStepIdx).isProducer();
```

Comment:
I might be mistaking, but AFAIK the call to [Schedule::filterConnection](/OPM/opm-simulators/blob/e15db6a89285b045ce7ed503591b9ccd698a74fd/ebos/eclcpgridvanguard.hh#L244) will remove non-local cells on all ranks besides 0. I assume that the intention here is to have a map of equal size on each process as otherwise the communication done below will not work. I doubt that this size requirement is really the case.

Or is my PR #2919 already merged?

Reply:
Indeed my memory failed me. It does not remove wells, but might produces ones without connections.
My problem with filter connection, that sparked this comment, was different.